common/flatpak-dir.c

```diff
@@ -4018,19 +4022,231 @@ repo_get_remote_collection_id (OstreeRepo *repo,
   return TRUE;
 }
 
+static const SecretSchema *
+get_api_server_access_token_schema (void)
+{
+    static const SecretSchema schema = {
+        "com.endlessm.HackAccountToken", SECRET_SCHEMA_NONE,
+        {
+            {  "serial", SECRET_SCHEMA_ATTRIBUTE_STRING },
+            {  "NULL", 0 },
+        }
+    };
+    return &schema;
+}
+
+static char *
+get_api_server_access_token (GCancellable  *cancellable,
```

Comment:
Would it make sense to store a server access token per API server? It feels awkward to me that we take an `api_server_url` in `get_repo_bearer_token` but we assume whatever API server we're talking to is going to use whatever token we're using.

Also +1 for using libsecret like this being a little too Hack-specific :) Perhaps we could split get_api_server_access_token out of flatpak-dir.c to make this a little easier to reason about, then upstream can have a naive implementation that uses a plain file per API server (probably the same scheme you use for cached tokens), while downstream can patch in the libsecret + serial number stuff for the Hack API server.